src/main/java/com/github/luben/zstd/ZstdInputStreamNoFinalizer.java

```diff
@@ -59,12 +59,12 @@ public ZstdInputStreamNoFinalizer(InputStream inStream) throws IOException {
      * @param inStream the stream to wrap
      * @param bufferPool the pool to fetch and return buffers
      */
-    public ZstdInputStreamNoFinalizer(InputStream inStream, BufferPool bufferPool) throws IOException {
+    public ZstdInputStreamNoFinalizer(InputStream inStream, BufferPool bufferPool) {
         super(inStream);
         this.bufferPool = bufferPool;
         this.srcByteBuffer = bufferPool.get(srcBuffSize);
         if (this.srcByteBuffer == null) {
-            throw new IOException("Cannot get ByteBuffer of size " + srcBuffSize + " from the BufferPool");
+            throw new ZstdException(Zstd.errMemoryAllocation(), "Cannot get ByteBuffer of size " + srcBuffSize + " from the BufferPool");
```

Comment:
The use case I have is that I want to differentiate between an error caused by a file that's not properly compressed vs. any other type of error. This looks like a case that's classified as one of those "any other type of error" cases. How would I as a user be expected to differentiate this case, would I inspect the Zstd error code from the exception?

Reply:
Don't we need to throw ZstdException where we are having `Zstd.isError()` check? 
Example: https://github.com/luben/zstd-jni/blob/master/src/main/java/com/github/luben/zstd/ZstdInputStreamNoFinalizer.java#L104

Reply:
Should we throw IllegalStateException or keep IOException in this specific case?

Reply:
You can get the error code with `ex.getErrorCode`. Even if we throw something else here, the above exception can still be thrown by the native library itself and it does not mean that there was something wrong with the compressed stream, just that there was some error in memory allocation.

Reply:
Makes sense. In that case, ship it!